File: backend/app/schemas/ai.py
from typing import Any

from pydantic import BaseModel, Field


class TextToDeckRequest(BaseModel):
    text: str = Field(..., min_length=10, max_length=50000, description="Input text to generate deck from")
    deck_title: str | None = Field(None, max_length=200, description="Optional title for the deck")


class TextToDeckResponse(BaseModel):
    deck_id: int
    title: str
    description: str | None
    cards_created: int
    message: str

File: frontend/src/services/ai.ts
import { api } from "./api";

export interface TextToDeckRequest {
  text: string;
  deck_title?: string | null;
}

export interface TextToDeckResponse {
  deck_id: number;
  title: string;
  description: string | null;
  cards_created: number;
  message: string;
}

export interface AIStatus {
  available: boolean;
  service: string;
}

export const aiService = {
  async generateDeckFromText(
    text: string,
    deckTitle?: string | null
  ): Promise<TextToDeckResponse> {
    const response = await api.post<TextToDeckResponse>("/ai/text-to-deck", {
      text,
      deck_title: deckTitle || null,
    });
    return response.data;
  },

  async getStatus(): Promise<AIStatus> {
    const response = await api.get<AIStatus>("/ai/status");
    return response.data;
  },
};
